### PR TITLE
add determinant

### PR DESCRIPTION
**Pull request by PBrdng**

### CODE REVIEW

File: src/det.jl [hunk 1]
```diff
@@ -0,0 +1,8 @@
+function Base.LinAlg.det(M::Matrix{<:MultivariatePolynomials.AbstractPolynomialLike})
```
Comment (saschatimme) [inline src/det.jl]: You only need `Matrix{<:AbstractPolynomialLike}` here

File: test/det.jl [hunk 2]
```diff
@@ -0,0 +1,6 @@
+@testset "Det" begin
+    Mod.@polyvar x y
+
+    @test det([x 1; y 1]) == x - y
+    @test det([x 1; x 1]) == 0
```
Comment (saschatimme) [inline test/det.jl]: This only tests the `else` clause, so maybe add a 5 by 5 matrix or so

File: src/det.jl (hunk 1 above)
Comment (saschatimme) [inline src/det.jl]: This doesn't work in 0.7: `Base.LinAlg.det`. Maybe import `det` (see comment above)

File: src/MultivariatePolynomials.jl [hunk 3]
```diff
@@ -8,7 +8,7 @@ using Compat
 import Compat.LinearAlgebra: dot, norm
```
Comment (saschatimme) [inline src/MultivariatePolynomials.jl]: also import `det` here (see below)

File: src/det.jl [hunk 4]
```diff
@@ -0,0 +1,8 @@
+function det(M::Matrix{<:AbstractPolynomialLike})
```
Comment (blegat) [inline src/det.jl]: I tend to prefer explicitely writing Compat.LinearAlgebra.det here and not doing the import. That is more readable as the information that is is tha Base det function and not a new one is not split in two places.
What do you think ?

Comment (saschatimme) [reply src/det.jl]: @blegat Sorry I merged a little bit too fast here. I addressed this in #87

Comment (blegat) [reply src/det.jl]: @saschatimme no problem ;)

**Pull request merged**